File: benchmarks/empty/c/bench.c
#include <stdio.h>
#include <stdlib.h>

void start_rapl();
void stop_rapl();

int main(int argc, char *argv[]) {
    int count = atoi(argv[1]);

    for (int i = 0; i < count; i++) {
        start_rapl();
        stop_rapl();
    }

    printf("C job done\n");

    return 0;
}

File: benchmarks/fibonacci sequence/bench.js
const runCount = 100
const fibParam = 47

function fib(n) {
    var a = 0, b = 1, t;
    while (n-- > 0) {
      t = a;
      a = b;
      b += t;
    }
    return a;
  }

const koffi = require('koffi');
const lib = koffi.load('target\\debug\\rapl_lib.dll');

const start = lib.func('int start_rapl()');
const stop = lib.func('void stop_rapl()');


for (let i = 0; i < runCount; i++){
    start();

    fib(fibParam);

    stop();
}

console.log("job done");  